src/services/databaseservice.cpp

```diff
@@ -243,6 +243,40 @@ bool DatabaseService::setupNoteFolderTables() {
         version = 11;
     }
 
+    if (version < 12) {
+        // create new tag table
+        queryDisk.exec("ALTER TABLE tag RENAME TO _tag");
+        queryDisk.exec("CREATE TABLE IF NOT EXISTS tag ("
```

Comment:
But I think you have created an index wrong, it has to be `CREATE UNIQUE INDEX IF NOT EXISTS idxUniqueTag ON tag (name COLLATE NOCASE, parent_id)` (see version 11). 

And could you please document why you did the migration at version 12 in the code, so it's clear why we didn't just `ALTER TABLE`.